src/vsr/replica.zig

```diff
@@ -689,6 +695,49 @@ pub fn ReplicaType(
                 self.superblock.working.free_set_reference(),
             ));
 
+            // TODO This can probably be performed concurrently to StateMachine.open().
+            self.client_sessions_checkpoint.open(
+                &self.grid,
+                self.superblock.working.client_sessions_reference(),
+                client_sessions_open_callback,
+            );
+        }
+
+        fn client_sessions_open_callback(client_sessions_checkpoint: *CheckpointTrailer) void {
+            const self = @fieldParentPtr(Self, "client_sessions_checkpoint", client_sessions_checkpoint);
+            assert(!self.state_machine_opened);
+            assert(self.commit_stage == .idle);
+            assert(self.syncing == .idle);
+            assert(self.sync_tables == null);
+            assert(self.grid_repair_tables.executing() == 0);
+            assert(std.meta.eql(
+                self.client_sessions_checkpoint.checkpoint_reference(),
+                self.superblock.working.client_sessions_reference(),
+            ));
+
+            {
+                const checkpoint = &self.client_sessions_checkpoint;
+                var address_previous: u64 = 0;
+                for (checkpoint.block_addresses[0..checkpoint.block_count()]) |address| {
+                    assert(address > 0);
+                    assert(address > address_previous);
+                    address_previous = address;
+                    self.grid.release(address);
+                }
+            }
+
+            const trailer_buffer = self.client_sessions_checkpoint.buffer;
+            const trailer_size = self.client_sessions_checkpoint.size;
+
+            self.client_sessions.reset();
+            if (self.superblock.working.client_sessions_reference().empty()) {
+                assert(trailer_size == 0);
+            } else {
+                assert(trailer_size == ClientSessions.encode_size_max);
```

Comment:
Maybe drop `_max` from `encode_size_max`? Otherwise, this line begs the question of whether it
was meant to be `<=` rather than `==`?